MediaBrowser.Model/Dlna/CodecProfile.cs

```diff
@@ -1,74 +1,108 @@
-
-        public ProfileCondition[] ApplyConditions { get; set; }
+        Conditions = [];
+        ApplyConditions = [];
+    }
 
-        [XmlAttribute("codec")]
-        public string Codec { get; set; }
+    /// <summary>
+    /// Gets or sets the <see cref="CodecType"/> which this container must meet.
+    /// </summary>
+    [XmlAttribute("type")]
+    public CodecType Type { get; set; }
 
-        [XmlAttribute("container")]
-        public string Container { get; set; }
+    /// <summary>
+    /// Gets or sets the list of <see cref="ProfileCondition"/> which this profile must meet.
+    /// </summary>
+    public ProfileCondition[] Conditions { get; set; }
 
-        [XmlAttribute("subcontainer")]
-        public string SubContainer { get; set; }
+    /// <summary>
+    /// Gets or sets the list of <see cref="ProfileCondition"/> to apply if this profile is met.
+    /// </summary>
+    public ProfileCondition[] ApplyConditions { get; set; }
 
-        public string[] GetCodecs()
-        {
-            return ContainerProfile.SplitValue(Codec);
-        }
+    /// <summary>
+    /// Gets or sets the codec(s) that this profile applies to.
+    /// </summary>
+    [XmlAttribute("codec")]
+    public string? Codec { get; set; }
 
-        private bool ContainsContainer(string container, bool useSubContainer = false)
-        {
-            var containerToCheck = useSubContainer && string.Equals(Container, "hls", StringComparison.OrdinalIgnoreCase) ? SubContainer : Container;
-            return ContainerProfile.ContainsContainer(containerToCheck, container);
-        }
+    /// <summary>
+    /// Gets or sets the container(s) which this profile will be applied to.
+    /// </summary>
+    [XmlAttribute("container")]
+    public string? Container { get; set; }
 
-        public bool ContainsAnyCodec(string codec, string container, bool useSubContainer = false)
-        {
-            return ContainsAnyCodec(ContainerProfile.SplitValue(codec), container, useSubContainer);
-        }
+    /// <summary>
+    /// Gets or sets the sub-container(s) which this profile will be applied to.
+    /// </summary>
+    [XmlAttribute("subcontainer")]
+    public string? SubContainer { get; set; }
 
-        public bool ContainsAnyCodec(string[] codec, string container, bool useSubContainer = false)
-        {
-            if (!ContainsContainer(container, useSubContainer))
-            {
-                return false;
-            }
+    /// <summary>
+    /// Checks to see whether the codecs and containers contain the given parameters.
+    /// </summary>
+    /// <param name="codecs">The codecs to match.</param>
+    /// <param name="container">The container to match.</param>
+    /// <param name="useSubContainer">Consider sub-containers.</param>
+    /// <returns>True if both conditions are met.</returns>
+    public bool ContainsAnyCodec(string[] codecs, string? container, bool useSubContainer = false)
```

Comment:
Why does this overload exist? And if it's for API compat, why doesn't it just call the ReadOnlyList overload?

Reply:
I removed it since it is now unused anyways.